cate/core/opimpl.py

```diff
@@ -383,9 +383,16 @@ def normalize_missing_time(ds: xr.Dataset) -> xr.Dataset:
     if not time_coverage_start and not time_coverage_end:
         # Can't do anything
         return ds
-
+    time = None
     if 'time' in ds:
-        time = ds.time
+        if isinstance(ds.time.values[0], datetime) or isinstance(ds.time.values[0], np.datetime64):
+            time = ds.time
+    elif 't' in ds:
```

Comment:
Save one line and one indent by concatening lines 390 and 391 with an 'and'

Reply:
No, I had this before with 'and' in one line, then it hit an error whenever time did not exist in ds, but it still tried to check if ds.time.values[0] was an instance of datetime. So I am keeping the current separation.